app/javascript/flavours/glitch/util/content_warning.js

```diff
@@ -0,0 +1,20 @@
+export function autoUnfoldCW (settings, status) {
+  if (!settings.getIn(['content_warnings', 'auto_unfold'])) {
+    return false;
+  }
+
+  const rawRegex = settings.getIn(['content_warnings', 'filter']);
+  let regex      = null;
+
+  try {
+    regex = rawRegex && new RegExp(rawRegex.trim(), 'i');
```

Comment:
Why not store it as a `RegExp` in Redux? That way you would avoid parsing the string for each status.

Reply:
Good point. I'm not too sure how they are serialized to local storage, and I think parsing the string before storage in the redux store would make the code more awkward. But I'll look into that, this could have some performance impact.